### PR TITLE
Bugfix/backend file 

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -1457,7 +1457,7 @@ def last_build_reason(repo, arch, package_name = nil)
       arch: arch
     )
 
-    data = Xmlhash.parse(xml_data.to_s)
+    data = Xmlhash.parse(xml_data.to_s.to_s)
```
Comment (coolo) [inline src/api/app/models/package.rb]: I think this is the ultimate sign that to_s just has the wrong name

File: src/api/lib/backend/file.rb [hunk 2]
```diff
@@ -55,8 +55,8 @@ def file(query = {})
       nil
     end
 
-    # Converts file into a String if it's valid
-    def to_s(query = {})
+    # Converts file into a String if it's valid and nil if it's not
```
Comment (Ana06) [inline src/api/lib/backend/file.rb]: Is this comment really needed? The method is pretty clear... :thinking:

File: src/api/lib/backend/file.rb [hunk 3]
```diff
@@ -55,8 +55,8 @@ def file(query = {})
...
+    def content(query = {})
       file(query)
       @file && valid? ? ::File.open(@file.path).read : nil
```
Comment (Ana06) [inline src/api/lib/backend/file.rb]: There should be a Rubocop cop to avoid using `nil` like this... :sob

**Comment by coolo** [comment]
Please don't merge this until #5140 is in - or you break the existing engine (as it's using .to_s)

**Comment by bgeuken** [comment]
@DavidKang @coolo I had to update a couple of places that were using the backend file class. Would you mind to have a second look?

File: src/api/app/models/project.rb [hunk 4]
```diff
@@ -145,14 +145,14 @@ def self.restore(project_name, backend_opts = {})
     project = Project.new(name: project_name)
 
     Project.transaction do
-      project.update_from_xml!(Xmlhash.parse(project.meta.to_s))
+      project.update_from_xml!(Xmlhash.parse(project.meta.content))
```
Comment (coolo) [inline src/api/app/models/project.rb]: is there any value in having meta not returning content right away? (just curious as you it .meta.content appears quite often in your PR)

Comment (bgeuken) [reply src/api/app/models/project.rb]: We would have to look closer into this. But you are right it seems that we only use the content right now.
The other thing is that there is a similar method called config and might be some others. I would keep them in sync.

**Comment by coolo** [comment]
One more test suite problem hidden by cassettes:

```
--- a/src/api/spec/controllers/webui/project_controller_spec.rb
+++ b/src/api/spec/controllers/webui/project_controller_spec.rb
@@ -1028,7 +1028,7 @@ RSpec.describe Webui::ProjectController, vcr: true do
 
     context 'Can not load project config' do
       before do
-        allow_any_instance_of(ProjectConfigFile).to receive(:to_s).and_return(nil)
+        allow_any_instance_of(ProjectConfigFile).to receive(:content).and_return(nil)
         get :prjconf, params: { project: apache_project }
       end
```

**Pull request merged**